### PR TITLE
rpm-deps: use qemu 8.0.0 and libvirt 9.5.0 plus fixes

### PR DESCRIPTION
**Pull request by victortoso**
**What this PR does / why we need it**:
At the moment, make rpm-deps fails with `Error: Package qemu-img-17:7.2.0-14.el9 does not exist`
Considering that 7.2.0 version is not present in [c9s/packages](https://mirror.stream.centos.org/9-stream/AppStream/x86_64/os/Packages/) - I thought it would make sense to update.

**Release note**:
```release-note
Updating the virt stack: QEMU 8.0.0, libvirt to 9.5.0, edk2 20230524,
passt 20230818, libguestfs and guestfs-tools 1.50.1, virtiofsd 1.7.2
```

### CODE REVIEW

**Comment by alicefr** [comment]
/approve
/hold

Please, unhold when the CI is green

**Comment by andreabolognani** [comment]
@victortoso we usually update all virtualization packages at the same time to roughly match what's going to be in the upcoming RHEL release, as that gives us better confidence that the various components have been tested together.

In this specific case, I would wait for libvirt 9.5.0 to be available from mirrors and include that in the update. The build has already happened, but IIRC it might take up to a couple of weeks to actually show up on the various mirrors...

If QEMU 7.2.0 no longer being available from mirrors is a blocking issue, we can perform the update in two steps, but otherwise I would do everything in one go.

**Comment by alicefr** [comment]
@victortoso @andreabolognani maybe instead of jumping to 8.x.x we could check if there is a more recent version for 7.2.x? If the updating isn't working anymore it is an issue

**Comment by victortoso** [comment]
jfyi, this is not a blocker to me. I can easily workaround it.

**Comment by alicefr** [comment]
/hold 
until we clarify how to upgrade

**Comment by andreabolognani** [comment]
> I don't see an intermediate version. andreabolognani do you know what regular centos stream users should expect? I mean right now they are getting a newer QEMU version independently from what KubeVirt does

@alicefr unfortunately there's no good answer to this in the context of CentOS Stream. It's a rolling release, and newer packages replace the older ones as time goes on.

For KubeVirt specifically, the fact that all packages that make their way to the WORKSPACE are mirrored to a separate cloud storage is a partial mitigation, meaning that building KubeVirt is not affected by this. Some development workflows still are.

> jfyi, this is not a blocker to me. I can easily workaround it.

@victortoso if you have a workaround, my preference would be to wait until libvirt 9.5.0 can be found on mirrors, and then do a big-bang refresh of the entire virtualization stack. I can take care of that in a week or so.

**Comment by victortoso** [comment]
> @victortoso if you have a workaround, my preference would be to wait until libvirt 9.5.0 can be found on mirrors, and then do a big-bang refresh of the entire virtualization stack. I can take care of that in a week or so.

That's fine by me. I'll close this one when you do the bulk update but If you happen to not have the time, let me know.

**Comment by andreabolognani** [comment]
> I see at least some koji builds for 9.5 Is libvirt 9.5 available now in centos stream?

Not yet, unfortunately :(

There was some issue with the build system and the 9.5.0-1 build couldn't make it all the way through the process. The maintainer has submitted a [9.5.0-2 build](https://kojihub.stream.centos.org/koji/buildinfo?buildID=34778) and that's looking a lot better already, but it's going to be at least a few days before it makes it into a CentOS Stream 9 compose.

I'm tracking this pretty closely though, and as soon as a 9.5.0 build is available I'll prepare a PR updating the entire virt stack.

**Comment by andreabolognani** [comment]
Quick update. The package maintainer for libvirt let me know that we're unfortunately not likely to get a usable 9.5.0 build until early August :(

**Comment by rmohr** [comment]
ok, given that we branche 1.0 and since we see that tests seem to work fine, I would vote for updating this PR once agian to use latest RPM packages and merge it as is and follow up later. We still have some months before we do the next release. @andreabolognani what do you think?

**Comment by victortoso** [comment]
I've updated the branch for now. I had to update [edk2](https://github.com/kubevirt/kubevirt/commit/d9d2b5935de89bf41b3d3f08673aefece2fb982a) because, like QEMU, the current version is not longer in c9s repos. Because I updated edk2,  [libvirt](https://github.com/kubevirt/kubevirt/commit/8fc65781e4148f7d0bd2a51c7eddb1abfbac1aa5) had to be updated to solve the conflict `libvirt-daemon-driver-qemu-0:9.0.0-3.el9 conflicts with edk2-aarch64-0:20230524-2.el9`.

**Comment by rmohr** [comment]
Hm. This looks like we need to modify a little bit more:

```
Unexpected Warning event received: testvmi-btpl8,c1d26c79-ff43-44e6-98a0-184fec965bc0: server error. command SyncVMI failed: "LibvirtError(Code=67, Domain=10, Message='unsupported configuration: itco model of watchdog is part of the machine and cannot have any address set.')"
```

**Comment by vasiliy-ul** [comment]
> Hm. This looks like we need to modify a little bit more:
> 
> ```
> Unexpected Warning event received: testvmi-btpl8,c1d26c79-ff43-44e6-98a0-184fec965bc0: server error. command SyncVMI failed: "LibvirtError(Code=67, Domain=10, Message='unsupported configuration: itco model of watchdog is part of the machine and cannot have any address set.')"
> ```

This is fixed here https://github.com/kubevirt/kubevirt/pull/10137

**Comment by andreabolognani** [comment]
@rmohr if there is nothing being blocked on virt packages being at the current versions I would personally prefer waiting and updating everything in one go once libvirt 9.5.0 is available. It avoids having to mirror a bunch of soon-to-be-obsoleted RPMs. But if the 1-2 weeks wait that's currently expected is too long, then feel free to go ahead with this PR and I'll take care of doing another update as a follow-up.

**Comment by victortoso** [comment]
/retest

So, just to clarify, this is not a blocker for me but I understand that anyone needing to update/add packages might hit this issue. If we want to move forward with this PR, I'd suggest having Vasiliy #10137 first to solve the underlying problem with watchdog. Up to you all.

**Comment by rmohr** [comment]
I think that takes a little bit too long to justify not being able to easily update packages on main during development.

@andreabolognani thanks for tracking this either way!

I think if we would do proper CVE tracking, we would also have to update on the release branches ...

**Comment by victortoso** [comment]
@rmohr should unhold https://github.com/kubevirt/kubevirt/pull/10137 as well? The tests here are going to fail without it

**Comment by rmohr** [comment]
> @rmohr should unhold #10137 as well? The tests here are going to fail without it

Right. Done. :+1:

**Comment by victortoso** [comment]
Both `1.26-sig-compute` and `1.27-sig-compute` are failing at:

```
 05:18:13: [sig-compute]Configurations VirtualMachineInstance definition [rfe_id:2262][crit:medium][vendor:cnv-qe@redhat.com][level:component]with EFI bootloader method [It] [Serial][test_id:4437]should enable EFI secure boot [sig-compute]
05:18:13: tests/vmi_configuration_test.go:656 
```

I'll investigate

**Comment by victortoso** [comment]
With current main branch
```
$ k create -f examples/vmi-secureboot.yaml
$ ./cluster-up/virtctl.sh console vmi-secureboot

[fedora@vmi-secureboot ~]$ dmesg | grep -i secureboot
[    0.000000] secureboot: Secure boot enabled
[    0.016676] secureboot: Secure boot enabled
[fedora@vmi-secureboot ~]$ sudo mokutil --sb-state
SecureBoot enabled
```

With this branch
```
$ k create -f examples/vmi-secureboot.yaml
$ ./cluster-up/virtctl.sh console vmi-secureboot

[fedora@vmi-secureboot ~]$ dmesg | grep -i secureboot
[    0.000000] secureboot: Secure boot disabled
[    0.016197] secureboot: Secure boot disabled
[fedora@vmi-secureboot ~]$ sudo mokutil --sb-state
SecureBoot disabled
Platform is in Setup Mode
```

Happy that we have functional tests. Some behavior changed that we might need to adapt/fix. I'm trying to understand what was it. cc @andreabolognani, in case you have a clue.

**Comment by rmohr** [comment]
Oh we ran into that recently: https://gitlab.com/libvirt/libvirt/-/issues/500

**Comment by rmohr** [comment]
/hold

let's stop CI for now.

**Comment by andreabolognani** [comment]
> Perhaps the best solution is to adapt to newer firmware selection?

Switching to feature-based firmware autoselection would almost certainly be the right move in the long run.

I think I tried looking into how much work that would be a while ago, but I remember being deterred by the fact that IIRC the path to the directory containing firmware files was passed around as an argument to virt-handler, which made me concerned about the potential ramifications.

Anyway, I'm currently working on libvirt patches that will address the issue, and I'm targeting them at RHEL 9.3 so they will be included in a future 9.5.0 build. However, I'm afraid such a build won't be available for a while longer...

**Comment by victortoso** [comment]
> Switching to feature-based firmware autoselection would almost certainly be the right move in the long run.

Okay, I'll start looking into it now. Also, I just saw that libvirt-9.5.0-5.el9.x86_64.rpm was included in c9s.

**Comment by andreabolognani** [comment]
> I just saw that libvirt-9.5.0-5.el9.x86_64.rpm was included in c9s.

Yup, I just noticed that too. At this point, it makes sense to incorporate that change into this PR and get everything in at the same time.

**Comment by victortoso** [comment]
> At this point, it makes sense to incorporate that change into this PR and get everything in at the same time.

Okay. The change to `firmware autoselection` can actually drop/change a bit more code but I added the bare minimal to see if we can get CI green first. I need to double check what chances it brings to configuration related to SEV, if any.

**Comment by vasiliy-ul** [comment]
> I need to double check what changes it brings to configuration related to SEV, if any.

@victortoso, FYI: SEV test job is not set as mandatory as of now. You can trigger it explicitly via

/test pull-kubevirt-e2e-k8s-1.26-sev

**Comment by brianmcarey** [comment]
> > I need to double check what changes it brings to configuration related to SEV, if any.
> 
> @victortoso, FYI: SEV test job is not set as mandatory as of now. You can trigger it explicitly via
> 
> /test pull-kubevirt-e2e-k8s-1.26-sev

The SEV node is not available at the moment - I am checking with AMD on the status of it.

**Comment by victortoso** [comment]
@andreabolognani, CI seems happy so far. Are you aware of any further issues related to moving from [legacy uefi template](https://gitlab.com/libvirt/libvirt/-/issues/500) to `firmware autoselection`? If not, `autoselection` seems to simplify a lot and I can work on doing some cleanups around EFI vars that we might not need anymore (in this PR or in a follow up).

The interesting commit is https://github.com/kubevirt/kubevirt/pull/10050/commits/274bdef1fca478ddb6a6c56eacb2ed0c23c2cb99

**Comment by brianmcarey** [comment]
The SEV node is available again.

/test pull-kubevirt-e2e-k8s-1.26-sev

**Comment by andreabolognani** [comment]
@victortoso I've taken a look at how KubeVirt currently implements firmware selection and tried to figure out whether switching to libvirt firmware autoselection would cause problems. I'll leave my thoughts here, in no particular order.

---

You're using

```xml
<os firmware='efi'>
  <loader secure='yes/no'/>
</os>
```

to request that Secure Boot is enabled/disabled, which is... Fine. The [current recommendations](https://libvirt.org/kbase/secureboot.html), however, involve enabling/disabling specific firmware features, such as

```xml
<os firmware='efi'>
  <firmware>
    <feature enabled='yes/no' name='enrolled-keys'/>
  </firmware>
</os>
```

Thanks to a bunch of compatibility code in libvirt, the end result is *roughly* the same, but if we're moving to firmware autoselection it would make sense to adopt the best practices.

I say *roughly* the same, because the `secure` attribute maps more accurately to the `secure-boot` firmware feature, which is however not necessarily the one you really want to control when you talk about enabling/disabling Secure Boot support. Some additional information about the topic can be found [here](https://libvirt.org/kbase/secureboot.html#additional-information). The takeaway is that `enrolled-keys=no` results in selecting `OVMF_CODE.secboot.fd` but keeping the Secure Boot feature disabled, while `secure-boot=no` results in selecting `OVMF_CODE.fd` which doesn't contain Secure Boot support at all.

It might be useful to compare this with your laptop's firmware settings. Turning off Secure Boot in the menu will obviously not result in a different EFI build, one with no Secure Boot support in it, being flashed on the spot :) Secure Boot support will still be present, but not active. This is the behavior you get with `enrolled-keys=no`.

---

Non-default firmware images, such as those needed on aarch64 and for SEV VMs, haven't been touched. We obviously need to ensure that those configurations keep working.

---

Currently the admission webhook has specific code that ensures that the SMM feature has been enabled when Secure Boot is requested, but firmware autoselection will automatically enable SMM in that case so the code should be dropped. Users will have one less think to consider when building the VM YAML, yay!

---

Speaking of the SEV use case, we're currently manually adding a symlink to `OVMF_CODE.cc.fd` via Bazel. With firmware autoselection, that's no longer needed or indeed desirable.

---

One thing that concerns me quite a bit is what happens to the NVRAM store between boots. The firmware autoselection process is driven by the contents and priority of JSON firmware descriptor files available on the system, and that can change between OS releases: as a concrete example, the latest build of edk2 has been changed so that, on aarch64, qcow2 firmware images are preferred to raw images.

libvirt will do its best to preserve the behavior of existing guests, but in the case of KubeVirt the VM is effectively defined from scratch before every single boot, at least based on my understanding. So libvirt could pick raw firmware for one boot, qcow2 firmware for the next, and suddenly the existing NVRAM store can no longer be parsed and the VM doesn't start.

Is the NVRAM store even preserved across boots? I've tried creating an EFI-enabled VMI and the directory where the files are stored was backed by an `emptyDir`, so it would seem that the NVRAM store gets thrown out every time? That would avoid the issue I was talking about, but also feels problematic for other reasons. What about the TPM's state, if one is configured? Is that also reset every single time?

---

Removing the check on the path of the NVRAM store is probably not a good idea. Even with autoselection enabled, we still want to make sure that libvirt generates the path we expect. Since that information is added at VM define time, however, the check will have to be moved from a unit test to a functional test.

---

The `OSFirmwareBios` constant doesn't serve any purpose AFAICT.

---

I'm sure I've missed a bunch of stuff. I'll add it when it pops back into my head :)

**Comment by xpivarc** [comment]
@victortoso @andreabolognani Do we need anything else here? Can we separate the auto selection from the PR?

**Comment by victortoso** [comment]
> @victortoso @andreabolognani Do we need anything else here? Can we separate the auto selection from the PR?

At the moment, updating libvirt without firmware auto selection would break secureboot [see this comment](https://github.com/kubevirt/kubevirt/pull/10050#issuecomment-1662060387). Although I haven't tested, [this patch series](https://listman.redhat.com/archives/libvir-list/2023-August/241171.html) should fix the problem and they were reviewed but not merged yet.

Not sure what the best course of action is:
1. Don't change to auto selection and wait to libvirt get the fixes in c9s (I'd guess two weeks or more)
2. Update to firmware auto selection with this PR (might take some days to [get it right](https://github.com/kubevirt/kubevirt/pull/10050#issuecomment-1675056312))
3. Don't update firmware auto selection with this PR (we drop this change for now) and do it in a different PR, breaking secureboot in the meanwhile.

I'd vote for 1 while doing the firmware auto selection in a different PR. If we get auto selection before the libvirt fixes, we should be able to move forward with updating rpms anyway.

**Comment by andreabolognani** [comment]
> At the moment, updating libvirt without firmware auto selection would break secureboot [see this comment](https://github.com/kubevirt/kubevirt/pull/10050#issuecomment-1662060387). Although I haven't tested, [this patch series](https://listman.redhat.com/archives/libvir-list/2023-August/241171.html) should fix the problem and they were reviewed but not merged yet.

It's been merged upstream now, and I've already prepared a [backport](https://gitlab.com/redhat/centos-stream/src/libvirt/-/merge_requests/7) targeting CentOS Stream 9.

> Not sure what the best course of action is:
> 
> 1. Don't change to auto selection and wait to libvirt get the fixes in c9s (I'd guess two weeks or more)
> 2. Update to firmware auto selection with this PR (might take some days to [get it right](https://github.com/kubevirt/kubevirt/pull/10050#issuecomment-1675056312))
> 3. Don't update firmware auto selection with this PR (we drop this change for now) and do it in a different PR, breaking secureboot in the meanwhile.
> 
> I'd vote for 1 while doing the firmware auto selection in a different PR. If we get auto selection before the libvirt fixes, we should be able to move forward with updating rpms anyway.

Option 1 would be my preferred one as well, as it's the one that carries the least risk of introducing regressions in KubeVirt. It will require a bit more patience than option 2 though. Option 3 is a complete no-go as far as I'm concerned.

I think investigating the switch to firmware autoselection in a separate PR in the meantime would be great.

**Comment by xpivarc** [comment]
Thanks @victortoso @andreabolognani,
I would be for the 1st option as well. Do we have any ETA?

**Comment by victortoso** [comment]
>  I would be for the 1st option as well. Do we have any ETA?

From the time we have a c9s build, I'd guess 2 weeks or so.

> I think investigating the switch to firmware autoselection in a separate PR in the meantime would be great.

I'll start now, thanks for the [comment](https://github.com/kubevirt/kubevirt/pull/10050#issuecomment-1675056312)!

**Comment by victortoso** [comment]
> If we get auto selection before the libvirt fixes, we should be able to move forward with updating rpms anyway.

Discussed quite a bit offline with @andreabolognani Today. One thing important to mention is that we should really introduce `firmware autoselection` **after** updating libvirt here, that is, waiting till the [libvirt build]( https://gitlab.com/redhat/centos-stream/src/libvirt/-/merge_requests/7) reaches centos stream 9.

I'll still be working on the `firmware autoselection` but the point is we shouldn't try to rush it.

**Comment by ormergi** [comment]
Hey, @victortoso looks like the new Libvirt build that include the EFI fix available in at https://kojihub.stream.centos.org/koji/buildinfo?buildID=36654
Is it possible to use it?

**Comment by andreabolognani** [comment]
> Hey, victortoso looks like the new Libvirt build that include the EFI fix (`libvirt-9.5.0-6.el9`)available in at https://kojihub.stream.centos.org/koji/buildinfo?buildID=36654
> https://kojihub.stream.centos.org/kojifiles/packages/libvirt/9.5.0/6.el9/
> Is it possible to use it?

Is there yum metadata available for that location? If not, bazeldnf will not be able to consume it.

Either way, it might not be a great idea to pick up builds that have not yet gone through the usual testing.

**Comment by victortoso** [comment]
> It might not be a great idea to pick up builds that have not yet gone through the usual testing.

I agree. Just a little longer and the rpms will be available for us to consume it.

**Comment by andreabolognani** [comment]
@ormergi the libvirt-9.5.0-6.el9 build which contains the fixes has passed verification yesterday, so it should show up in an internal compose soon. Hopefully within a day or two?

Then there's an additional delay before packages reach the CentOS Stream mirrors. This could take a few more days, in the worst case scenario even a week or longer I think. Unfortunately I don't have much visibility into how often these batch processes are run

 tl;dr we're not quite there yet I'm afraid, but at least progress is being made. I'm sorry for the continued pain this situation is causing, but there's really nothing I can do to speed things along :(

**Comment by victortoso** [comment]
libvirt-9.5.0-6 is now available in centos stream. I'll be updating this PR shortly.

**Comment by andreabolognani** [comment]
@victortoso looks good overall, but you're also going to want

```patch
diff --git a/hack/rpm-deps.sh b/hack/rpm-deps.sh
index 74a372428..fecbdf820 100755
--- a/hack/rpm-deps.sh
+++ b/hack/rpm-deps.sh
@@ -10,10 +10,10 @@ LIBVIRT_VERSION=${LIBVIRT_VERSION:-0:9.5.0-6.el9}
 QEMU_VERSION=${QEMU_VERSION:-17:8.0.0-13.el9}
 SEABIOS_VERSION=${SEABIOS_VERSION:-0:1.16.1-1.el9}
 EDK2_VERSION=${EDK2_VERSION:-0:20230524-3.el9}
-LIBGUESTFS_VERSION=${LIBGUESTFS_VERSION:-1:1.48.4-4.el9}
-GUESTFSTOOLS_VERSION=${GUESTFSTOOLS_VERSION:-0:1.48.2-8.el9}
+LIBGUESTFS_VERSION=${LIBGUESTFS_VERSION:-1:1.50.1-6.el9}
+GUESTFSTOOLS_VERSION=${GUESTFSTOOLS_VERSION:-0:1.50.1-3.el9}
 PASST_VERSION=${PASST_VERSION:-0:0^20230818.g0af928e-4.el9}
-VIRTIOFSD_VERSION=${VIRTIOFSD_VERSION:-0:1.5.0-1.el9}
+VIRTIOFSD_VERSION=${VIRTIOFSD_VERSION:-0:1.7.2-1.el9}
 SWTPM_VERSION=${SWTPM_VERSION:-0:0.8.0-1.el9}
 SINGLE_ARCH=${SINGLE_ARCH:-""}
 BASESYSTEM=${BASESYSTEM:-"centos-stream-release"}
```

to fully bring the virtualization stack up to date.

I've noticed that this adds a number of new dependencies, specifically adobe-source-code-pro-fonts which looks a bit out of place. I haven't spent time figuring out which package exactly drags this in, and whether we can avoid it.

Only tangentially related, but we also need

```patch
diff --git a/hack/rpm-deps.sh b/hack/rpm-deps.sh
index 74a372428..96e0fe16e 100755
--- a/hack/rpm-deps.sh
+++ b/hack/rpm-deps.sh
@@ -261,6 +261,8 @@ if [ -z "${SINGLE_ARCH}" ] || [ "${SINGLE_ARCH}" == "x86_64" ]; then
         --name pr-helper_x86_64 \
         --basesystem ${BASESYSTEM} \
         ${bazeldnf_repos} \
+        $centos_main \
+        $centos_extra \
         $pr_helper

     # remove all RPMs which are no longer referenced by a rpmtree
@@ -362,6 +364,8 @@ if [ -z "${SINGLE_ARCH}" ] || [ "${SINGLE_ARCH}" == "aarch64" ]; then
         --name pr-helper_aarch64 --arch aarch64 \
         --basesystem ${BASESYSTEM} \
         ${bazeldnf_repos} \
+        $centos_main \
+        $centos_extra \
         $pr_helper

     # remove all RPMs which are no longer referenced by a rpmtree
```

to avoid dependency flipping for the pr-helper image. This gets rid of the full version of coreutils (we want coreutils-single instead) and a couple glibc-langpack-* packages. You can either have this change as a fix early in your series, or I can take care of it separately as a follow-up.

Additional nits:

* having the specific `make rpm-deps` error listed in each commit message is IMO unnecessary. The more generic note that you have in the libvirt commit is more than enough;
* in the release notes section of the PR description, the bit about firmware autoselection doesn't apply to the current version and so it should be dropped;
* again in the release notes, please spell libvirt without a capital letter ;)

**Comment by victortoso** [comment]
@andreabolognani thanks for the quick review. I've added passt, libguestfs, guestfs-tools and virtiofsd. Double checked the change you suggested for pr-helper. All looks fine from my POV.

> I've noticed that this adds a number of new dependencies, specifically adobe-source-code-pro-fonts which looks a bit out of place. I haven't spent time figuring out which package exactly drags this in, and whether we can avoid it.

I'll start looking at it now. I'd say we can reduce any extra dependency as a follow up.

**Comment by andreabolognani** [comment]
> I'll start looking at it now. I'd say we can reduce any extra dependency as a follow up.

Agreed, the amount of additional disk space because of the extra packages is relatively small and it only affects the guestfs image anyway, so it's not affecting every VM. This update is long overdue, so let's not delay it any further unless doing so is absolutely necessary.

/lgtm

**Comment by victortoso** [comment]
/retest-required

```
13:30:15: Summarizing 1 Failure:
13:30:15:   [FAIL] [Serial][sig-compute]InfrastructuredownwardMetrics [It] [test_id:6535]should be published to a vmi and periodically updated [sig-compute]
13:30:15:   tests/infrastructure/downward-metrics.go:53 

....

12:16:32:   [FAILED] Expected success, but got an error:
12:16:32:       <*errors.errorString | 0xc003494090>: 
12:16:32:       send to spawned process command reached the timeout 9.974840705s
12:16:32:       {
12:16:32:           s: "send to spawned process command reached the timeout 9.974840705s",
12:16:32:       } 

```

**Comment by victortoso** [comment]
/retest-required

```
17:21:13: Summarizing 1 Failure:
17:21:13:   [FAIL] [Serial][sig-compute]InfrastructuredownwardMetrics [It] [test_id:6535]should be published to a vmi and periodically updated [sig-compute]
17:21:13:   tests/infrastructure/downward-metrics.go:53

....

16:07:36:   [FAILED] Expected success, but got an error:
16:07:36:       <*errors.errorString | 0xc005160000>: 
16:07:36:       send to spawned process command reached the timeout 9.979019564s
16:07:36:       {
16:07:36:           s: "send to spawned process command reached the timeout 9.979019564s",
16:07:36:       } 
```

**Comment by victortoso** [comment]
@alicefr, @xpivarc and @rmohr, I think this is ready for another look/unhold?

No problems with secureboot. All the virt stack is up to date with latest centos stream 9.

**Comment by rmohr** [comment]
/approve

/unhold

Thanks for following up on this over such a long time. 🎉

**Comment by vasiliy-ul** [comment]
Let's also run the SEV test job

/test pull-kubevirt-e2e-k8s-1.26-sev

**Comment by xpivarc** [comment]
Thanks @victortoso @andreabolognani !!!

**Comment by andreabolognani** [comment]
I'll just point out something that I hadn't noticed until now, because it got sort of drowned in the `make rpm-deps` noise: with this update, the legacy monolithic `libvirtd` daemon is no longer included in the `virt-launcher` image!

KubeVirt had already switched to the modular `virtqemud` last year with https://github.com/kubevirt/kubevirt/pull/8619, but up until now the libvirt RPM packages were not flexible enough to accomodate a `libvirtd`-less deployment. Well, now they are :)

As part of this move `virtproxyd` and `virtlockd`, which KubeVirt never used, are gone too. All in all, a nice cleanup :)
